react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,50 @@
+import { useState, useEffect, useCallback } from 'react';
+import { FaArrowLeft, FaArrowRight } from 'react-icons/fa';
+import { ImageCard } from './ImageCard';
+import { Dots } from './Dots';
+import './Carousel.css';
+
+type Image = {
+  src: string;
+  alt: string;
+};
+
+type CarouselProps = {
+  images: Image[];
+};
+
+export function Carousel({ images }: CarouselProps) {
+  const [activeIndex, setActiveIndex] = useState(0);
+
+  const nextImage = useCallback(() => {
+    setActiveIndex((prevIndex) => (prevIndex + 1) % images.length);
+  }, [images.length]);
```

Comment:
The dependency should be `images` rather than just `images.length`. We don't necessarily care if the length changes, but we do care if the `images` change